tools/ci-trigger/pr.go

```diff
@@ -151,6 +158,41 @@ func (p *pullRequest) createBuild(ctx context.Context, buildClient *cloudbuild.S
 				}
 			}
 		}
+	physicalDeviceLoop:
+		for i, physicalDevice := range p.Physical {
+			if physicalDevice.Type == d {
+				if len(physicalDevice.Tests) == 0 {
+					continue
+				}
+				for _, v := range physicalDevice.Tests {
+					if v.Status != "pending authorization" {
+						continue physicalDeviceLoop
+					}
+				}
+				jobID, err := uuid.NewRandom()
+				if err != nil {
+					return fmt.Errorf("uuid.NewRandom device %q: %w", physicalDevice.Type.String(), err)
+				}
+				p.Physical[i].ArchivePath = objPath
```

Comment:
isn't p.Physical[i] the same as physicalDevice here and below?

Reply:
It is but there were no pointers being used so modifying physicalDevice directly wouldn't work.  I updated this and think it is easier to follow now.